Public/Src/Cache/MemoizationStore/Library/Sessions/LocalCacheConfiguration.cs

```diff
@@ -0,0 +1,87 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics.ContractsLight;
+using System.Linq;
+using System.Security;
+using System.Text;
+using System.Threading.Tasks;
+using BuildXL.Cache.ContentStore.Interfaces.FileSystem;
+using BuildXL.Cache.ContentStore.Sessions;
+using BuildXL.Cache.ContentStore.Stores;
+using BuildXL.Cache.Host.Configuration;
+using BuildXL.Cache.Host.Service;
+
+namespace BuildXL.Cache.MemoizationStore.Sessions
+{
+    /// <summary>
+    /// Cache configuration for <see cref="LocalCache"/>.
+    /// </summary>
+    public class LocalCacheConfiguration
+    {
+        /// <summary>
+        /// Whether the cache will communicate with a server in a separate process via GRPC.
+        /// </summary>
+        public bool EnableContentServer { get; }
+
+        /// <summary>
+        /// The GRPC port to use.
+        /// </summary>
+        public int GrpcPort { get; }
+
+        /// <summary>
+        /// Name of one of the named caches owned by CASaaS.
+        /// </summary>
+        public string CacheName { get; }
+
+        /// <summary>
+        /// Name of the custom scenario that the CAS services.
+        /// Allows multiple CAS services to coexist in a machine
+        /// since this factors into the cache root and the event that
+        /// identifies a particular CAS instance.
+        /// </summary>
+        public string ScenarioName { get; }
+
+        /// <nodoc />
+        public uint RetryIntervalSeconds { get; }

```

Comment:
nit: should we always use int instead of uint?

Reply:
Sure! But I won't update any of our public surface from `uint` to `int`.

---
In reply to: [280152542](https://github.com/Microsoft/BuildXL/pull/220#discussion_r280152542) [](ancestors = 280152542)

Reply:
I thought that this is the new stuff, so it means that it should not be a problem. Right?

Reply:
I should have been more specific. I won't update the constructor for `ServiceClientContentStore` from `uint` to `int`, because it is used by AzDev.